src/every.test.ts

```diff
@@ -0,0 +1,107 @@
+import { every } from "./every";
+import { isNumber } from "./isNumber";
+import { pipe } from "./pipe";
+
+describe("runtime", () => {
+  describe("data-first", () => {
+    test("returns correct values on simple arrays", () => {
+      const trueInput = [1, 2, 3];
+      const trueData = every(trueInput, (val) => val < 5);
+      expect(trueData).toBe(true);
+
+      const falseInput = ["foo", "bar", "buzz"];
+      const falseData = every(falseInput, (val) => val.length > 3);
+      expect(falseData).toBe(false);
+    });
+
+    // empty set check
+    test("returns true on empty arrays regardless of predicate", () => {
+      const input = [] as const;
+      const data = every(input, () => false);
+      expect(data).toBe(true);
+    });
+
+    test("works with indexed predicate", () => {
+      const input = [1, 2, 3];
+      const data = every(input, (_, index) => index < 5);
+      expect(data).toBe(true);
```

Comment:
👍🏼 nice, this is an advantage of moving indexed predicates to the main signature, we test them like regular logic and not disconnected from other concerns!